### PR TITLE
[GITHUB-13] [GITHUB-14] Update to Latest Standards

### PR DESCRIPTION
**Pull request by dennisconrad**
* Also support PHP 7.2

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1,41 +1,32 @@
...
-    # removed due to doctrine trying to use memcache or redis by default to cache
-    - php-redis
-    - php-memcache
-  paths:
-    etc: /etc/php7.0
-    fpm_pool: /etc/php7.0/fpm/pool.d
+sansible_php_version: php7.0
+sansible_php_extras: []
+sansible_php_pecl: []
+sansible_php_modules:
```
Comment (lobsterdore) [inline defaults/main.yml]: I actually think we should reduce this list right down, I don't think this list make sense to be honest :), let's just install curl by default as pretty much everyone needs that:

```YAML
sansible_php_modules:
  - "{{ sansible_php_version }}-curl"
```

File: defaults/main.yml [hunk 2]
```diff
@@ -1,41 +1,32 @@
...
-    # E: Unable to locate package php7.0-mssql
-    # - php7.0-mssql
-    - php7.0-sqlite3
-    # removed due to doctrine trying to use memcache or redis by default to cache
-    - php-redis
-    - php-memcache
-  paths:
-    etc: /etc/php7.0
-    fpm_pool: /etc/php7.0/fpm/pool.d
+sansible_php_version: php7.0
```
Comment (lobsterdore) [inline defaults/main.yml]: Can you alpha order all of these parameters?

File: meta/main.yml [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
 allow_duplicates: yes
-dependencies: [ ]
+dependencies: []
 
 galaxy_info:
   author: Wojtek Oledzki
   description: PHP and FPM instalation
   license: MIT
-  min_ansible_version: 2.0
+  min_ansible_version: 2.2
```
Comment (lobsterdore) [inline meta/main.yml]: Can you add specific versions to the platform list?

File: molecule/default/playbook.yml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+  pre_tasks:
+    - name: Ensure nginx Directories Exist
+      file:
+        path: /etc/nginx/includes/
+        state: directory
+
+  roles:
+    - role: php
+      sansible_php_version: php7.2
```
Comment (lobsterdore) [inline molecule/default/playbook.yml]: Need to be alpha ordered.

File: molecule/default/playbook.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+---
+
+- name: Converge
+  hosts: all
+
+  pre_tasks:
+    - name: Ensure nginx Directories Exist
```
Comment (lobsterdore) [inline molecule/default/playbook.yml]: Perhaps we should include the Sansible nginx role as a dependency? Might be better than munging a directory. You could include the users and groups role as well so we don't have to use the php user for fpm.

File: tasks/build.yml [hunk 6]
```diff
@@ -4,35 +4,39 @@
...
-    - "{{ php.version }}-cli"
+    - "{{ sansible_php_version }}"
+    - "{{ sansible_php_version }}-common"
+    - "{{ sansible_php_version }}-fpm"
+    - "{{ sansible_php_version }}-cli"
+
+- name: Don't Install mcrypt Module if PHP Version is 7.2
+  set_fact:
+    sansible_php_modules: "{{ sansible_php_modules | difference(['php7.2-mcrypt']) }}"
+  when: sansible_php_version == "php7.2"
```
Comment (lobsterdore) [inline tasks/build.yml]: I don't think we will need this anymore since we are removing mcrypt as a default.

File: tasks/main.yml [hunk 7]
```diff
@@ -1,5 +1,13 @@
 ---
 
+- name: Assert Required Variables
+  assert:
+    that:
+      - sansible_php_fpm_description is defined
+      - sansible_php_fpm_user is defined
+      - sansible_php_fpm_group is defined
+      - sansible_php_fpm_chroot is defined
```
Comment (lobsterdore) [inline tasks/main.yml]: These variables can be undefined if someone wants to install PHP without FPM, perhaps these should be move to the when clause for the fpm.yml include in build.yml.

File: templates/init.j2 [hunk 8]
```diff
@@ -1,10 +1,10 @@
-# php5-fpm-{{ php.fpm.user }} - The PHP FastCGI Process Manager
+# php5-fpm-{{ sansible_php_fpm_user }} - The PHP FastCGI Process Manager
```
Comment (lobsterdore) [inline templates/init.j2]: I think we can remove this file since we are no longer installing it.

File: meta/main.yml [hunk 9]
```diff
@@ -1,16 +1,18 @@
 ---
 
 allow_duplicates: yes
-dependencies: [ ]
+dependencies:
+  - sansible.nginx
```
Comment (lobsterdore) [inline meta/main.yml]: I don't think this should be here, we only want Nginx when running the tests, I would suggest having something like a test Galaxy requirements file that we add use via Molecule (or perhaps a test deps target in Makefile), the we add invoke the role in molecule/default/playbook.yml so it affects the tests only.

Comment (dennisconrad) [reply meta/main.yml]: I will revert to just creating the directories then.  I'm just testing whether the packages have been installed and the configuration files have been created anyway.

**Comment by lobsterdore** [comment]
LGTM 👍

**Pull request merged**